ddtrace/tracer/remote_config.go

```diff
@@ -18,6 +20,38 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
 )
 
+type dynamicInstrumentationRCProbeConfig struct {
+	runtimeID            string
+	productConfigPath    string
+	productConfigContent string
```

Comment:
Nit: even if we don't switch to a map, let's rename to `configPath` and `configContent`? I don't think `product` adds any value.